### PR TITLE
Sources table server side pagination

### LINKED ISSUE #1294: Sources page crashes when trying to sort by save date
bug-5

+ Replace text "Loading sources" with a spinner.

### PR DESCRIPTION
**Pull request by acrellin**
source_list_pagination

group_sources_refactored

Closes #1294 
Closes https://github.com/skyportal/skyportal/issues/1285

### CODE REVIEW

**Comment by acrellin** [comment]
@guynir42 @kmshin1397 this might still need a test or two to be updated, but otherwise is ready for review

**Comment by acrellin** [comment]
Actually, this is green (at least on GA) and ready for review.

**Comment by guynir42** [comment]
It looks very good. I assume that other stuff like sorting and filtering will come in a later PR?

**Comment by acrellin** [comment]
> It looks very good. I assume that other stuff like sorting and filtering will come in a later PR?

@guynir42 that's correct, that won't be part of this PR.

**Review by guynir42 — APPROVED** [review]

File: static/js/ducks/sources.js [hunk 1]
```diff
@@ -5,87 +5,133 @@ export const FETCH_SOURCES = "skyportal/FETCH_SOURCES";
...
+  action
+) => {
   switch (action.type) {
     case FETCH_SOURCES: {
       return {
         ...state,
-        queryInProgress: action.parameters.body.pageNumber === undefined,
+        latest: {
+          ...state.latest,
+          queryInProgress: action.parameters.body.pageNumber === undefined,
```
Comment (kmshin1397) [inline static/js/ducks/sources.js:73]: Do we need the `queryInProgress` state? Doesn't seem to be used anywhere for sources or group sources

File: static/js/components/SourceTable.jsx [hunk 2]
```diff
@@ -482,6 +510,15 @@ const SourceTable = ({ sources, title, sourceStatus = "saved", groupID }) => {
...
     selectableRows: "none",
+    sort: false,
+    onTableChange: handleTableChange,
+    serverSide: true,
+    rowsPerPage: numPerPage,
+    page: pageNumber - 1,
+    rowsPerPageOptions: [10, 25, 50, 75, 100, 200],
+    jumpToPage: true,
+    pagination: true,
+    count: totalMatches,
```
Comment (kmshin1397) [inline static/js/components/SourceTable.jsx:521]:
```suggestion
    count: totalMatches,
    filter: false,
    search: false,
```
Filtering and searching actually breaks when you enable serverSide, and you need to set up custom handlers/renders to get them to work again. I suggest we disable these for now and add that functionality back in through a future PR.

**Review by kmshin1397 — CHANGES_REQUESTED** [review]
Couple of minor changes, but looking good! I think the refactor made all this much cleaner.

**Comment by kmshin1397** [comment]
Oh, and also the merge conflict - I think I introduced it by making that unsaved sources PR this morning. Seems like you included similar logic in here and I didn't notice. You can probably choose either implementation when resolving the conflict. I think they're functionally equivalent.

File: static/js/ducks/sources.js (hunk 1 above)
Comment (acrellin) [reply static/js/ducks/sources.js:73]: You're right it's not needed for the groupSources branches, but is still needed in the "latest" (non-group-specific) state for the SourceList logic. I'll remove them from the group sources reducers

File: static/js/components/SourceTable.jsx (hunk 2 above)
Comment (acrellin) [reply static/js/components/SourceTable.jsx:521]: :+1:

Comment (acrellin) [reply static/js/components/SourceTable.jsx:521]: Done

File: static/js/ducks/sources.js (hunk 1 above)
Comment (acrellin) [reply static/js/ducks/sources.js:73]: Done

**Review by kmshin1397 — APPROVED** [review]

**Comment by kmshin1397** [comment]
Awesome, will merge after CI runs

**Comment by acrellin** [comment]
@kmshin1397 I think I botched something in the merge conflict resolution -- will get it sorted asap

**Comment by kmshin1397** [comment]
Right yeah I just noticed the failing linter checks on GA. I'll hold off on merging until it's fixed

**Review by kmshin1397 — APPROVED** [review]

**Pull request merged**